### DISCUSSION TITLE
Make sisu-equinox more generic?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-tycho/tycho#1156] [Ideas]
As the name suggest `sisu-equinox` is dedicated to equinox, but in fact many of its API is not really specific to equinox, one might even argue that as launching and behavior of OSGi Frameworks is well defined it should be possible to completely abstract that Equinox is used.

I therefore like to suggest to rename `sisu-equinox` to `sisu-osgi` and as part of this move all generic api to one module (currently it is split over `sisu-equinox-api` / `sisu-equinox-embedder` / `sisu-equinox-launching`.

Also `sisu-equinox-launching` currently is tied to `tycho-embedder-api` by `org.eclipse.tycho.ArtifactDescriptor` and `org.eclipse.tycho.ArtifactKey` this seems not suitable and should be replaced with a sisu interface (that Tycho maybe implements/extends). When we have do the API split we can easily have a `sisu-equinox-launching` that contains the equinox specific parts.

This will make the reuse of this components in other context much more the 3.x release seems a good opportunity for this, any ideas / concerns?

**Comment by akurtakov** [comment]
I don't have concerns over such a change. Only question is do you plan to work on it soon? I think it's time to set a date for Tycho 3.0 and work towards it. Smth like end of September is what is currently on my mind.

**Reply by laeubi** [reply]
Yes, I'll start with it immediately, my main goal is to to move Tycho to the OSGi-Framework-Connect (see #1152) and it seems better be placed inside `sisu-equinox` (but won't be equinox specific!) and could be interesting for others as well and simplify so much things, I even think in the long term a dedicated `sisu-osgi` would make sense, but for now we could incubate this at Tycho I think.

**Comment by laeubi** [comment]
@mickaelistria any concerns? Do we know about any other users of `sisu-equinox` we might want to notify beside the usual release notes?

**Reply by mickaelistria** [reply]
No I have no concern about the rename or whatever. It seems like an improvement overall.
My only concern, if I were the one to implement it would be "is it worth the effort? is it worth taking higher priority than X,Y,Z?" and my answer would probably be "No, I have higher priority stuff". But if you have a perception that your answer is "yes, it's worth it", then feel free to move forward, it's welcome.
